### PR TITLE
Br invoice items relationships & invoices relationships

### LINKED ISSUE #33: Relationship Endpoint - invoice transactions
* returns a collection of associated transactions

GET /api/v1/invoices/:id/transactions

### LINKED ISSUE #34: Relationship Endpoint - invoice invoice_items
* returns a collection of associated invoice items

GET /api/v1/invoices/:id/invoice_items

### LINKED ISSUE #35: Relationship Endpoint - invoice items
* returns a collection of associated items

GET /api/v1/invoices/:id/items

### LINKED ISSUE #36: Relationship Endpoint - invoice customer
* returns the associated customer

GET /api/v1/invoices/:id/customer

### LINKED ISSUE #37: Relationship Endpoint - invoice merchant
* returns the associated merchant

GET /api/v1/invoices/:id/merchant

### LINKED ISSUE #51: Relationship Endpoint - invoice_items invoices
* returns the associated invoice

GET /api/v1/invoice_items/:id/invoice

### LINKED ISSUE #52: Relationship Endpoint - invoice_items items
* returns the associated item

GET /api/v1/invoice_items/:id/item

### PR DESCRIPTION
**Pull request by riverswb**
@lucyconklin @case-eee

Adds all of the endpoints for invoices relationships
Adds all endpoints for invoice items relationships
Tweaks to some factories
All tests passing
19 failures on spec helper
closes #33
closes #34
closes #35
closes #36
closes #37
closes #51 
closes #52

### CODE REVIEW

File: spec/requests/api/v1/relationships/invoice_items_relationships_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    get "/api/v1/invoice_items/#{@invoice_item.id}/invoice"
+
+    found = JSON.parse(response.body)
+
+    expect(found).to be_a(Hash)
+    expect(found["id"]).to eq(@invoice_item.invoice_id)
+    expect(found).to have_key("merchant_id")
+    expect(found).to have_key("customer_id")
+    expect(found).to have_key("status")
+  end
```
Comment (lucyconklin) [inline spec/requests/api/v1/relationships/invoice_items_relationships_spec.rb:19]: I've been wondering if maybe we could clean up these tests by testing;
expect(found.keys).to eq(["id","merchant_id","customer_id","status"])

And test all of them in one line

Comment (riverswb) [reply spec/requests/api/v1/relationships/invoice_items_relationships_spec.rb:19]: That would definitely clean things up, I like it!

Comment (case-eee) [reply spec/requests/api/v1/relationships/invoice_items_relationships_spec.rb:19]: 👍

**Review by case-eee — COMMENTED** [review]
@riverswb @lucyconklin keep cruising along 👍

**Pull request merged**